install.sh

```diff
@@ -605,220 +597,192 @@ prepare_environment_for_buildah() {
 
     # Check the sysctl value: user.max_user_namespaces
-    USER_NAMESPACES_SETTING=$(sysctl -n user.max_user_namespaces)
-    if [ "$USER_NAMESPACES_SETTING" -lt 15000 ]; then
+    USER_NAMESPACES_SETTING="$(sysctl -n user.max_user_namespaces)"
+    if [[ "$USER_NAMESPACES_SETTING" < "15000" ]]; then
       echo 'user.max_user_namespaces = 15000' | run_as_root "tee -a /etc/sysctl.conf"
       run_as_root "sysctl -p"
       log::info "Updated user.max_user_namespaces value"
     fi
+  else 
+    log::warn "sysctl not found"
   fi
 }
 
-# Function to install on Arch Linux
-install_arch() {
-  local package="$1"
-  run_as_root "pacman -S $package"
-}
-
-# Function to install on CentOS
-install_centos() {
-  local package="$1"
-  run_as_root "yum -y install $package"
-}
-
-# Function to install on Debian
-install_debian() {
-  local package="$1"
-  run_as_root "apt-get update"
-  run_as_root "apt-get -y install $package"
-}
-
-# Function to install on Fedora
-install_fedora() {
-  local package="$1"
-  run_as_root "dnf -y install $package"
+get_linux_distro() {
+  declare distro
+  if [[ -f /etc/os-release ]]; then
+    . "/etc/os-release"
+    distro="$NAME"
+  elif type lsb_release 1>/dev/null 2>&1; then
+    distro="$(lsb_release -si)"
+  elif [[ -f "/etc/lsb-release" ]]; then
+    . "/etc/lsb-release"
+    distro="$DISTRIB_ID"
+  elif [[ -f "/etc/debian_version" ]]; then
+    distro="debian"
+  elif [[ -f "/etc/SuSe-release" ]]; then
+    distro="suse"
+  elif [[ -f "/etc/redhat-release" ]]; then
+    distro="redhat"
+  fi
+  echo "$distro"
 }
 
-# Function to install on Fedora SilverBlue
-install_fedora_silverblue() {
+install_package() {
   local package="$1"
-}
+  distro=$(get_linux_distro)
+  
+  log::info "Detected Linux distribution: $distro"
 
-# Function to install on Fedora CoreOS
-install_fedora_coreos() {
-  local package="$1"    
-  rpm-ostree install $package
+  case "$distro" in
+    *CentOS*|*Red\ Hat*|*Fedora*)
+      run_as_root "yum -y install $package"
+      ;;
+    *Debian*|*Ubuntu*)
+      run_as_root "apt-get -y update"
+      run_as_root "DEBIAN_FRONTEND=noninteractive apt-get -y install $package"
+      ;;
+    *SUSE*|*openSUSE*)
+      run_as_root "zypper install $package"
+      ;;
+    *RHEL*)
+      if [[ "$distro" == *"7"* ]]; then
+        run_as_root "yum -y install $package"
+      elif [[ "$distro" == *"8"* ]]; then
+        run_as_root "dnf module install -y $package"  
+      fi
+      ;;
+    *)
+      abort "Unsupported distribution: $distro"
+      ;;
+  esac
 }
 
-# Function to install on Gentoo
-install_gentoo() {
-  run_as_root "emerge app-containers/buildah"
-}
+install_git() {
+  distro=$(get_linux_distro)
+  
+  # Установка зависимостей в зависимости от дистрибутива
+  log::info "Installing dependencies for Git compilation..."
+  
+  case "$distro" in
+    *Debian*|*Ubuntu*)
+      install_package "make gcc libssl-dev zlib1g-dev libcurl4-openssl-dev gettext autoconf automake libexpat1-dev perl"
```

Comment:
Oh, you sure we want to compile git, isn't it a bit too much? Maybe, if the git version available is too old, we would try to install it from some third-party repos instead, like [here](https://askubuntu.com/a/568596)?